carddrawer/src/test/java/com/meli/android/carddrawer/model/CardAnimatorTest.kt

```diff
@@ -0,0 +1,175 @@
+    private lateinit var cardBackLayoutMock: View
+
+    private fun initialize() {
+        cardAnimator = CardAnimator(contextMock, cardFrontLayoutMock, cardBackLayoutMock)
+    }
+
+    @Test
+    fun `when getting cardBackLayout of CardAnimator then return value passed in the constructor`() {
+        initialize()
+        val cardBackLayout = cardAnimator.getDeclaredField("cardBackLayout")
+        Assert.assertEquals(cardBackLayoutMock, cardBackLayout)
+    }
+
+    @Test
+    fun `when getting cardFrontLayout of CardAnimator then return value passed in the constructor`() {
+        initialize()
+        val cardFrontLayout = cardAnimator.getDeclaredField("cardFrontLayout")
+        Assert.assertEquals(cardFrontLayoutMock, cardFrontLayout)
+    }
+
+    @Test
+    fun `when getting default card color of CardAnimator then return value not null`() {
+        initialize()
+        val defaultCardColor = cardAnimator.getDeclaredField("defaultCardColor")
+        Assert.assertNotNull(defaultCardColor)
+    }
+
+    @Test
+    fun `when getting context of CardAnimator then return value not null`() {
+        initialize()
+        val context = cardAnimator.getDeclaredField("context")
+        Assert.assertNotNull(context)
+    }
+
+    @Test
+    fun `when call function switchView with POSITION_FRONT per parameter then fill showingLayout with the same value`() {
+        initialize()
+        val showingLayout = cardAnimator.getDeclaredField("showingLayout")
+        cardAnimator.switchView(FieldPosition.POSITION_FRONT)
+        Assert.assertEquals(showingLayout, FieldPosition.POSITION_FRONT)
+    }
+
+    @Test
+    fun `when call function switchView with 3 per parameter then fill showingLayout with POSITION_FRONT`() {
+        initialize()
+        val showingLayout = cardAnimator.getDeclaredField("showingLayout")
+        cardAnimator.switchView(3)
+        Assert.assertEquals(showingLayout, FieldPosition.POSITION_FRONT)
+    }
+
+    @Test
+    fun `when call function switchView with POSITION_BACK per parameter then fill showingLayout with the same value`() {
+        val mock = mockk<CardAnimator>(relaxed = true)
+        every { mock.switchView(FieldPosition.POSITION_BACK) } answers { callOriginal() }
+        mock.switchView(FieldPosition.POSITION_BACK)
+        val showingLayout = mock.getDeclaredField("showingLayout")
+        Assert.assertEquals(showingLayout, FieldPosition.POSITION_BACK)
+    }
+
+    @Test
+    fun `when call function switchViewWithoutAnimation with POSITION_FRONT per parameter fill showingLayout with POSITION_FRONT`() {
+        initialize()
+        cardAnimator.switchViewWithoutAnimation(FieldPosition.POSITION_FRONT)
+        val showingLayout = cardAnimator.getDeclaredField("showingLayout")
+        Assert.assertEquals(showingLayout, FieldPosition.POSITION_FRONT)
+    }
+
+    @Test
+    fun `when call function switchViewWithoutAnimation with POSITION_BACK per parameter fill showingLayout with POSITION_FRONT`() {
```

Comment:
in this sentence, is missing "then" sentence.

Reply:
adjusted